### PR TITLE
Implement five `_mm_cmpistr`

### LINKED ISSUE #308: Implement SSE4.2 text processing intrinsics
[Apache Impala](https://impala.apache.org/) adopts SSE2NEON along with partial SSE4.2 text processing intrinsics support in file [be/src/util/sse-util.h](https://github.com/apache/impala/blob/master/be/src/util/sse-util.h):
* cmpestrm
* cmpestri

However, the above were neither optimized nor generic. Instead, we can provide the full-functioned instrinsics in SSE2NEON.

### PR DESCRIPTION
**Pull request by Cuda-Chen**
Implement `_mm_cmpistra`, `_mm_cmpistrc`, `_mm_cmpistro`, `_mm_cmpistrs`, and `_mm_cmpistrz`.

Close #308.

### CODE REVIEW

File: sse2neon.h [hunk 1]
```diff
@@ -9139,34 +9139,51 @@ FORCE_INLINE int _mm_cmpestrz(__m128i a,
...
+            uint16x8_t equal_mask_##str = vreinterpretq_u16_u8(          \
+                vceqq_u8(vreinterpretq_u8_m128i(str), vdupq_n_u8(0)));   \
+            uint8x8_t res_##str = vshrn_n_u16(equal_mask_##str, 4);      \
+            uint64_t matches_##str =                                     \
+                vget_lane_u64(vreinterpret_u64_u8(res_##str), 0);        \
+            len = _sse2neon_ctzll(matches_##str) >> 2;                   \
+        }                                                                \
     } while (0)
 
+#define SSE2NEON_CMPISTRX_GET_LEN(a, b, la, lb)     \
```
Comment (jserv) [inline sse2neon.h:9161]: It is confusing to have `SSE2NEON_CMPISTRX_GET_LEN` and `SSE2NEON_CMPISTRX_GET_STR_LEN` since they look similar. Can you clarify?

Comment (Cuda-Chen) [reply sse2neon.h:9161]: `SSE2NEON_CMPISTRX_GET_LEN` is used to get the length of string `a` and `b`.
While `SSE2NEON_CMPISTRX_GET_STR_LEN` is used to get the length of one string (`_mm_cmpistrs` and `_mm_cmpistrz` just need to get the length of string `a` and `b`, respectively).

Comment (jserv) [reply sse2neon.h:9161]: The name `SSE2NEON_CMPISTRX_GET_STR_LEN` contains two `STR` identifiers. It looks a bit weird. We can rename `SSE2NEON_CMPISTRX_GET_LEN` to `SSE2NEON_CMPISTRX_LEN_PAIR`, which retrieves the length of two strings.

Comment (jserv) [reply sse2neon.h:9161]: Similarly, `SSE2NEON_CMPISTRX_LENGTH` looks slightly understandable.

Comment (Cuda-Chen) [reply sse2neon.h:9161]: Sounds great. I will make another commit to include this suggestion.

Comment (Cuda-Chen) [reply sse2neon.h]: Done.

**Pull request merged**